etl/steps/data/meadow/wb/2025-01-15/poverty_projections.py

```diff
@@ -0,0 +1,60 @@
+"""Load a snapshot and create a meadow dataset."""
+
+from etl.helpers import PathFinder, create_dataset
+
+# Get paths and naming conventions for current step.
+paths = PathFinder(__file__)
+
+# Define files directory
+FILES_DIRECTORY = "FR_WLD_2024_198/Reproducibility package/Chapter 1/1-data/raw/forecasts"
+
+# Define index columns
+INDEX_COLUMNS = ["country", "year", "povertyline", "scenario"]
+
+# Define table parameters
+TABLE_PARAMETERS = {
+    "country": {"file": "FGTcountry_1990_2050_3pr24.dta"},
+    "region": {"file": "FGTregion_1990_2050_3pr24.dta"},
+    "global": {"file": "FGTglobal_1990_2050_3pr24.dta"},
+}
+
+
+def run(dest_dir: str) -> None:
+    #
+    # Load inputs.
+    #
+    # Retrieve snapshot.
+    snap = paths.load_snapshot("reproducibility_package_poverty_prosperity_planet.zip")
+
+    # Define empty list to store tables.
+    tables = []
+    for table, table_config in TABLE_PARAMETERS.items():
+        # Load data from snapshot.
+        tb = snap.read_in_archive(f"{FILES_DIRECTORY}/{table_config['file']}")
```

Comment:
Hi @lucasrodes! Do you know why this is failing? I only updated the version of the data and it has the same structure, but now it fails with
```
ValueError: File is neither a zip nor a tar file.
```

Reply:
I was testing this, and found no error on my end. Can you provide the complete traceback?

FWIW, I've added some new features the `Snapshot` class, so that we can read multiple files from an archive more efficiently. So far, we had to unarchive the file multiple times (one for each read operation)

Reply:
Thanks! Yes, let me know when https://github.com/owid/etl/pull/3863 is ready to go! (I am following the PR as well)

Reply:
merged! @paarriagadap